crates/solidity/outputs/cargo/slang_solidity/src/generated/bindings/generated/binding_rules.rs

```diff
@@ -405,24 +406,32 @@ attribute symbol_reference = symbol  => type = "push_symbol", symbol = symbol, i
   edge typeof -> @type_name.output
 }
 
-@function [FunctionDefinition parameters: [ParametersDeclaration
-    [Parameters @param item: [Parameter]]
-]] {
-  edge @param.lexical_scope -> @function.lexical_scope
+@params [ParametersDeclaration] {
+  node @params.lexical_scope
+  node @params.defs
+}
+
+@params [ParametersDeclaration [Parameters @param item: [Parameter]]] {
+  edge @param.lexical_scope -> @params.lexical_scope
+  edge @params.defs -> @param.def
+}
+
+@function [FunctionDefinition @params parameters: [ParametersDeclaration]] {
+  edge @params.lexical_scope -> @function.lexical_scope
 
   ;; Input parameters are available in the function scope
-  edge @function.lexical_scope -> @param.def
-  attr (@function.lexical_scope -> @param.def) precedence = 1
+  edge @function.lexical_scope -> @params.defs
+  attr (@function.lexical_scope -> @params.defs) precedence = 1
 }
 
 @function [FunctionDefinition returns: [ReturnsDeclaration
-    [ParametersDeclaration [Parameters @param item: [Parameter]]]
+    @return_params [ParametersDeclaration]
```

Comment:
is this change to return params a bug fix? not sure I understand how that affects the existing behaviour.

Should we add unit tests to cover that as well?

Reply:
It's a refactor in order to be able to reuse the processing of `ParametersDeclaration` for try statements as well. Before this change, the `ParametersDeclaration` would not have any graph nodes associated and the parameters were connected to the rest of the graph through the `FunctionDefinition` itself. With this change, there are intermediate scoped nodes (`lexical_scope` and `defs`) to connect both to the parameters and to the function definition. But we can now connect the nodes associated to the parameters also in the `try` statement.